tensorflow/compiler/mlir/tosa/tests/lower-complex-types.mlir

```diff
@@ -40,3 +40,26 @@ func.func @test_mixed_output(%arg0: tensor<1x4x4x2xf32>, %arg1: tensor<1x4x4xf32
   %0 = builtin.unrealized_conversion_cast %arg0 : tensor<1x4x4x2xf32> to tensor<1x4x4xcomplex<f32>>
   return %0, %arg1 : tensor<1x4x4xcomplex<f32>>, tensor<1x4x4xf32>
 }
+
+// -----
+
+// CHECK-LABEL: test_complex_input_output_op
+// CHECK-SAME: %[[VAL_0:.*]]: tensor<1x4x4x2xf32>
+// CHECK: %[[VAL_1:.*]] = "tosa.slice"(%[[VAL_0]]) {size = array<i64: 1, 4, 4, 1>, start = array<i64: 0, 0, 0, 0>} : (tensor<1x4x4x2xf32>) -> tensor<1x4x4x1xf32>
+// CHECK: %[[VAL_2:.*]] = "tosa.reshape"(%[[VAL_1]]) {new_shape = array<i64: 1, 4, 4>} : (tensor<1x4x4x1xf32>) -> tensor<1x4x4xf32>
+// CHECK: %[[VAL_3:.*]], %[[VAL_4:.*]] = "tosa.rfft2d"(%[[VAL_2]]) : (tensor<1x4x4xf32>) -> (tensor<1x4x3xf32>, tensor<1x4x3xf32>)
+// CHECK: %[[VAL_5:.*]] = "tosa.reshape"(%[[VAL_3]]) {new_shape = array<i64: 1, 4, 3, 1>} : (tensor<1x4x3xf32>) -> tensor<1x4x3x1xf32>
+// CHECK: %[[VAL_6:.*]] = "tosa.reshape"(%[[VAL_4]]) {new_shape = array<i64: 1, 4, 3, 1>} : (tensor<1x4x3xf32>) -> tensor<1x4x3x1xf32>
+// CHECK: %[[VAL_7:.*]] = "tosa.concat"(%[[VAL_5]], %[[VAL_6]]) {axis = 3 : i64} : (tensor<1x4x3x1xf32>, tensor<1x4x3x1xf32>) -> tensor<1x4x3x2xf32>
+// CHECK: return %[[VAL_7]] : tensor<1x4x3x2xf32>
+func.func @test_complex_input_output_op(%arg0: tensor<1x4x4xcomplex<f32>>) -> (tensor<1x4x3xcomplex<f32>>) {
+    %0 = builtin.unrealized_conversion_cast %arg0 : tensor<1x4x4xcomplex<f32>> to tensor<1x4x4x2xf32>
```

Comment:
It's a little bit weird to see an unrealized conversion cast here, normally they only occur as part of dialect conversion and not before or after the conversion. Is there a way for the user to perform this transform?

Reply:
SGTM.